tests/unit/RetroGenerationService.unitTest.ts

```diff
@@ -203,6 +204,49 @@ describe("RetroGenerationService", () => {
 
       });
     });
+
+    context("with custom defects", () => {
+      it("should show custom defects in Failure/Advisory, QA & I,comments column", () => {
+        const testResult = JSON.parse(testResultCustomDefectsResponse.body)[0];
+        const myTests: ITestResults[] = [];
+        for (let i = 0; i < 20; i++) {
+          myTests.push(testResult);
+        }
+        const mockTestResultsService = jest.fn().mockImplementation(() => {
+          return {
+            getTestResults: () => Promise.resolve(TestResultsService.prototype.expandTestResults(myTests))
+          };
+        });
+        const mockActivitiesService = jest.fn().mockImplementation(() => {
+          return {
+            getActivities: () => Promise.resolve([])
+          };
+        });
+        const retroGenSvc = new RetroGenerationService(new mockTestResultsService(), new mockActivitiesService());
+        return retroGenSvc.generateRetroReport(activity)
+          .then((result: any) => {
+            const workbook = new Excel.Workbook();
+            const stream = new Duplex();
+            stream.push(result.fileBuffer); // Convert the incoming file to a readable stream
+            stream.push(null);
+
+            return workbook.xlsx.read(stream)
+              .then((excelFile: any) => {
+                const reportSheet: Excel.Worksheet = excelFile.getWorksheet(1);
+
+                expect(excelFile.creator).toEqual("Commercial Vehicles Services Beta Team");
+                expect(excelFile.company).toEqual("Drivers and Vehicles Standards Agency");
+                expect(reportSheet.name).toEqual("Retrokey report");
+
+                expect(reportSheet.getCell("M17").value).toContain("Custom Defects:");
+                // tslint:disable-next-line
+                // expect(reportSheet.getCell("B17").value).toEqual("Test");
+                // expect(reportSheet.getCell("B36").value).toEqual("Test");
+                // expect(reportSheet.getCell("B38").value).toEqual("Document Requests");
```

Comment:
Why are these lines commented out? The unit test should be checking that each of the cells gets calculated correctly, not just that the header cell has the text Custom Defects in it.

Reply:
Its a copy paste issue. `M17` isn't pointing to the header. Its pointing to the data in the Comments/Defects column. and `Custom Defects:` is appended as a string in that column